src/audio/igo_nr/igo_nr.c

```diff
@@ -0,0 +1,804 @@
+	case SOF_IPC_FRAME_S24_4LE:
+		comp_info(dev, "set_capture_func(), SOF_IPC_FRAME_S24_4LE");
+		cd->igo_nr_func = igo_nr_capture_s24;
+		break;
+#endif
+#if CONFIG_FORMAT_S32LE
+	case SOF_IPC_FRAME_S32_LE:
+		comp_info(dev, "set_capture_func(), SOF_IPC_FRAME_S32_LE");
+		cd->igo_nr_func = igo_nr_capture_s32;
+		break;
+#endif
+	default:
+		comp_err(dev, "set_capture_func(), invalid frame_fmt");
+		return -EINVAL;
+	}
+	return 0;
+}
+
+static struct comp_dev *igo_nr_new(const struct comp_driver *drv,
+				   struct sof_ipc_comp *comp)
+{
+	struct sof_ipc_comp_process *ipc_igo_nr =
+		(struct sof_ipc_comp_process *)comp;
+	struct comp_dev *dev;
+	struct comp_data *cd = NULL;
+	size_t bs = ipc_igo_nr->size;
+	int32_t ret;
+
+	comp_cl_info(&comp_igo_nr, "igo_nr_new()");
+
+	/* Check first that configuration blob size is sane */
+	if (bs > SOF_IGO_NR_MAX_SIZE) {
+		comp_cl_err(&comp_igo_nr, "igo_nr_new() error: configuration blob size = %u > %d",
+			    bs, SOF_IGO_NR_MAX_SIZE);
+		return NULL;
+	}
+
+	dev = comp_alloc(drv, COMP_SIZE(struct sof_ipc_comp_process));
+	if (!dev)
+		return NULL;
+
+	memcpy_s(COMP_GET_IPC(dev, sof_ipc_comp_process),
+		 sizeof(struct sof_ipc_comp_process), ipc_igo_nr,
+		 sizeof(struct sof_ipc_comp_process));
+
+	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*cd));
+	if (!cd)
+		goto err;
+
+	ret = IgoLibGetInfo(&cd->igo_lib_info);
+	if (ret != IGO_RET_OK) {
+		comp_cl_err(&comp_igo_nr, "igo_nr_new(): IgoLibGetInfo() Failed.");
+		goto err;
+	}
+
+	cd->p_handle = rballoc(0, SOF_MEM_CAPS_RAM, cd->igo_lib_info.handle_size);
+	if (!cd->p_handle) {
+		comp_cl_err(&comp_igo_nr, "igo_nr_new(): igo_handle memory rballoc error");
+		goto err;
+	}
+
+	comp_set_drvdata(dev, cd);
+
+	/* Handler for configuration data */
+	cd->model_handler = comp_data_blob_handler_new(dev);
+	if (!cd->model_handler) {
+		comp_cl_err(&comp_igo_nr, "igo_nr_new(): comp_data_blob_handler_new() failed.");
+		goto err;
+	}
+
+	/* Get configuration data */
+	ret = comp_init_data_blob(cd->model_handler, bs, ipc_igo_nr->data);
+	if (ret < 0) {
+		comp_cl_err(&comp_igo_nr, "igo_nr_new(): comp_init_data_blob() failed.");
+		goto err;
+	}
+	dev->state = COMP_STATE_READY;
+
+	comp_cl_info(&comp_igo_nr, "igo_nr created");
+
+	return dev;
+
+err:
+	rfree(dev);
+	rfree(cd);
+	return NULL;
+}
+
+static void igo_nr_free(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	comp_info(dev, "igo_nr_free()");
+
+	comp_data_blob_handler_free(cd->model_handler);
+
+	rfree(cd->p_handle);
+	rfree(cd);
+	rfree(dev);
+}
+
+static int32_t igo_nr_verify_params(struct comp_dev *dev,
+				    struct sof_ipc_stream_params *params)
+{
+	int32_t ret;
+
+	comp_dbg(dev, "igo_nr_verify_params()");
+
+	/* update downstream (playback) or upstream (capture) buffer parameters
+	 */
+	ret = comp_verify_params(dev, BUFF_PARAMS_RATE, params);
+	if (ret < 0) {
+		comp_err(dev, "igo_nr_verify_params(): comp_verify_params() failed.");
+		return ret;
+	}
```

Comment:
i'd remove the return ret here and just return ret outside the if

Reply:
Agreed. Just moved outside.